aws-android-sdk-iot/src/main/java/com/amazonaws/mobileconnectors/iot/AWSIotMqttManager.java

```diff
@@ -808,7 +831,16 @@ private void mqttConnect(MqttConnectOptions options) {
         options.setKeepAliveInterval(userKeepAlive);
 
         if (isMetricsEnabled()) {
-            options.setUserName("?SDK=Android&Version=" + SDK_VERSION);
+            StringBuilder username = new StringBuilder("?SDK=Android&Version=" + SDK_VERSION);
+
+            if (usernameFields != null) {
+                // Append each of the user-specified key-value pair to the username field for the connection
+                for (Map.Entry<String, String> usernameField : usernameFields.entrySet()) {
```

Comment:
what happens if the key or value is empty?

Reply:
We pass key and/or value passed by user as is. Constraint validation will be performed by the service and exceptions, if any, will be propagated to the user.

Reply:
Added the tests